### PR TITLE
Add prediction utils that enable inferer

### PR DESCRIPTION
**Pull request by Can-Zhao**
Signed-off-by: Can Zhao <canz@nvidia.com>

Fixes # .

### Description
Add prediction utils that enable inferer

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/apps/detection/networks/retinanet_predictor.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+from typing import Dict, List, Optional
+
+import torch
+from torch import Tensor, nn
+
+from monai.inferers import SlidingWindowInferer
+
+
+def convert_dict_value_to_list(head_outputs: Dict[str, List[Tensor]]) -> Dict[str, List[Tensor]]:
```
Comment (Nic-Ma) [inline monai/apps/detection/networks/retinanet_predictor.py:20]: Should we make all these module functions in this function to be private with prefix "_"? I think they are only used by this `DictPredictor`?

Thanks.

**Review by wyli — APPROVED** [review]
utils for wrapping the multi-head inferer

**Pull request merged**